src/modules/Remote.rb

```diff
@@ -131,54 +148,29 @@ def Propose
     # Read the current status
     # @return true on success
     def Read
-      # are the proper services enabled in xinetd?
-      xinetd_conf = SCR.Read(path(".etc.xinetd_conf.services")) || {}
-      xinetd_vnc1_enabled = xinetd_conf.any? { |c| c["service"] == "vnc1" && c["enabled"] }
-
-      log.info "VNC1 enabled: #{xinetd_vnc1_enabled}"
+      display_manager_remote_access = SCR.Read(path(".sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS")).to_s == "yes"
 
+      xdm        = Service.Enabled(XDM_SERVICE_NAME)
       vncmanager = Service.Enabled(VNCMANAGER_SERVICE)
-      vnc_service_running = xinetd_vnc1_enabled || vncmanager
+      xvnc       = Service.Enabled(XVNC_SERVICE)
+      xvnc_novnc = Service.Enabled(XVNC_NOVNC_SERVICE)
+
+      if xdm && display_manager_remote_access && (xvnc || vncmanager)
+        if xvnc
+          Enable()
+        else
+          EnableVncManager()
+        end
 
-      if xinet_ready? && vnc_service_running
-        Enable() if xinetd_vnc1_enabled
-        EnableVncManager() if vncmanager
+        EnableWebVnc() if xvnc_novnc
       else
         Disable()
+        DisableWebVnc()
```

Comment:
how should I know from the above condition that just web vnc should be disabled?

Reply:
I am not sure if I understand your question.
Web VNC depends on VNC. If the conditions for VNC are not fulfilled, Web VNC is not enabled either.